proto/concept.proto

```diff
@@ -0,0 +1,963 @@
+            Thing.SetHas.Res thing_set_has_res = 102;
+            Thing.UnsetHas.Res thing_unset_has_res = 103;
+
+            Relation.AddPlayer.Res relation_add_player_res = 200;
+            Relation.RemovePlayer.Res relation_remove_player_res = 201;
+        }
+    }
+
+    message ResPart {
+        oneof res {
+            Thing.GetHas.ResPart thing_get_has_res_part = 101;
+            Thing.GetRelations.ResPart thing_get_relations_res_part = 104;
+            Thing.GetPlaying.ResPart thing_get_playing_res_part = 105;
+
+            Relation.GetPlayers.ResPart relation_get_players_res_part = 202;
+            Relation.GetPlayersByRoleType.ResPart relation_get_players_by_role_type_res_part = 203;
+            Relation.GetRelating.ResPart relation_get_relating_res_part = 204;
+
+            Attribute.GetOwners.ResPart attribute_get_owners_res_part = 300;
+        }
+    }
+
+    message Delete {
+        message Req {}
+        message Res {}
+    }
+
+    message GetHas {
+        message Req {
+            oneof filter {
+                AttributeTypes attribute_types = 1;
+                Annotations annotation_filter = 2;
+            }
+            // Only one filter can be set at a time. repeated can't be used in a oneof
+            message AttributeTypes {
+                repeated AttributeType attribute_types = 1;
+            }
+            message Annotations {
+                repeated Type.Annotation annotations = 1;
+            }
+        }
+        message ResPart {
+            repeated Attribute attributes = 1;
+        }
+    }
+
+    message SetHas {
+        message Req {
+            Attribute attribute = 1;
+        }
+        message Res {}
+    }
+
+    message UnsetHas {
+        message Req {
+            Attribute attribute = 1;
+        }
+        message Res {}
+    }
+
+    message GetRelations {
+        message Req {
+            repeated RoleType role_types = 1;
+        }
+        message ResPart {
+            repeated Relation relations = 1;
+        }
+    }
+
+    message GetPlaying {
+        message Req {}
+        message ResPart {
+            repeated RoleType role_types = 1;
+        }
+    }
+}
+
+message Entity {
+    bytes iid = 1;
+    EntityType entity_type = 2;
+    bool inferred = 3;
+}
+
+message Relation {
+    bytes iid = 1;
+    RelationType relation_type = 2;
+    bool inferred = 3;
+
+    message AddPlayer {
+        message Req {
+            RoleType role_type = 1;
+            Thing player = 2;
+        }
+        message Res {}
+    }
+
+    message RemovePlayer {
+        message Req {
+            RoleType role_type = 1;
+            Thing player = 2;
+        }
+        message Res {}
+    }
+
+    message GetPlayers {
+        message Req {
+            repeated RoleType role_types = 1;
+        }
+        message ResPart {
+            repeated Thing things = 1;
+        }
+    }
+
+    message GetPlayersByRoleType {
+        message RoleTypeWithPlayer {
```

Comment:
Let's turn this into `Relation.RolePlayer` and replace the usages of their pair in all of the Relation sub-messages that match exactly (`AddPlayer.Request, RemovePlayer.Reques`- these become renamed `AddRolePlayer` and `RemoveRolePlayer`) etc.